fsharp-backend/src/LibExecution/StdLib/LibList.fs

```diff
@@ -45,59 +45,51 @@ let fns : List<BuiltInFn> =
-//     ; fn =
-//
-//           (function
-//           | _, [DList l] ->
-//             ( match List.hd l with
-//             | Some dv ->
-//                 Dval.to_opt_just dv
-//             | None ->
-//                 DOption OptNothing )
-//           | args ->
-//               incorrectArgs ())
-//     ; sqlSpec = NotYetImplementedTODO
-//       ; previewable = Pure
-//     ; deprecated = NotDeprecated }
-//   ; { name = fn "List" "tail" 0
-//
-//     ; parameters = [Param.make "list" TList]
-//     ; returnType = TOption
-//     ; description =
-//         "If the list contains at least one value, returns `Just` a list of every value other than the first. Otherwise, returns `Nothing`."
-//     ; fn =
-//         (* This matches Elm's implementation, with the added benefit that the error rail
-//          * means you don't need to handle unwrapping the option
-//          * unless the passed list is truly empty (which shouldn't happen in most practical uses). *)
-//
-//           (function
-//           | _, [DList l] ->
-//             ( match List.tl l with
-//             | Some dv ->
-//                 DList dv |> Dval.to_opt_just
-//             | None ->
-//                 DOption OptNothing )
-//           | args ->
-//               incorrectArgs ())
-//     ; sqlSpec = NotYetImplementedTODO
-//       ; previewable = Pure
-//     ; deprecated = NotDeprecated }
-//   ; { name = fn "List" "empty" 0
-//
-//     ; parameters = []
-//     ; returnType = TList
-//     ; description = "Returns an empty list."
-//     ; fn =  (function _, [] -> DList [] | args -> incorrectArgs ())
-//     ; sqlSpec = NotYetImplementedTODO
-//       ; previewable = Pure
-//     ; deprecated = NotDeprecated }
+      deprecated = ReplacedBy(fn "List" "head" 2) }
+    { name = fn "List" "head" 2
+      parameters = [ Param.make "list" (TList varA) "" ]
+      returnType = varA
+      description =
+        "Returns `Just` the head (first value) of a list. Returns `Nothing` if the list is empty."
+      fn =
+        (function
+        | _, [ DList l ] ->
+            (match List.tryHead l with
+             | Some dv -> Value(l.Head)
+             | None -> Value(DOption None))
+        | args -> incorrectArgs ())
+      sqlSpec = NotYetImplementedTODO
+      previewable = Pure
+      deprecated = NotDeprecated }
+    { name = fn "List" "tail" 0
+      parameters = [ Param.make "list" (TList varA) "" ]
+      returnType = TList varA
+      description =
+        "If the list contains at least one value, returns `Just` a list of every value other than the first. Otherwise, returns `Nothing`."
+      fn =
+        (* This matches Elm's implementation, with the added benefit that the error rail
+         * means you don't need to handle unwrapping the option
+         * unless the passed list is truly empty (which shouldn't happen in most practical uses). *)
+        (function
+        | _, [ DList l ] ->
+            (match List.tryLast l with
```

Comment:
TryLast is very slow, it recurses to the end of the list. I think you could use `if List.length l > 0` instead.